service/dynamodb/dynamodbattribute/fields_go1.9.go

```diff
@@ -0,0 +1,23 @@
+// +build go1.9
+
+package dynamodbattribute
+
+import "sync"
+
+var fieldCache fieldCacher
+
+type fieldCacher struct {
+	cache sync.Map
```

Comment:
Would like to know if I am thinking correctly about this use case where sync.map chosen ahead of maps with RWMutex!

I see the con for sync.Map to not be type safe, and would involve more reflection to type assert. But it has better performance when higher number of cores are used. 

Are we using sync.Map as we have might have concurrent looping for frequent lookups on fieldCache?

Reply:
One of the benefits of sync.Map is that it fits the use case of the (un)marshaler's write reflected field set once, read often, since the fields of a structure will never change while an application.

Adding the `fieldsCacher` type helps with the typing since its helper methods provide wrap direct access to the Map and provide type safety.

Reply:
Yeah, I agree that lack of type safety is a minus, but like @jasdel mentioned, wrapping access to the underlying map makes it nicer.  My initial crack at this used `sync.RWMutex`+`map[reflect.Type]field`, but I ended up swapping over to `sync.Map` after looking through some of the different stdlib encoding packages that have reflection caches.